### DISCUSSION TITLE
Change in UsersService.update: working code now says "Primary key is missing in update payload"

### DISCUSSION THREAD

**Discussion by gdpdils** [directus/directus#5879] [Archive]
Hi, I just updated my app from Directus `9.0.0-rc.59` to Directus `9.0.0-rc.69` and noticed a change in the behaviour of `UsersService.update`. Specifically, the following code

```javascript
	await userService.update(
		{
			aad_access_token: aadUser.access_token,
			aad_refresh_token: aadUser.refresh_token,
		},
		user[0].id
	);
```

used to work, while now it rises an exception saying that the `Primary key is missing in update payload.`. Changing the code to

```javascript
	await userService.update(
		{
                        id: user[0].id,
			aad_access_token: aadUser.access_token,
			aad_refresh_token: aadUser.refresh_token,
		},
		user[0].id
	);
```

(so, explicitly adding `id` to the payload) fixes the problem. Is this a bug or the expected result and hence the way the api is going to work from now on? If so, is this the new behaviour for all services' `update` methods? Thank you.

**Comment by rijkvanzanten** [comment]
`update` has been replaced by `updateOne`, `updateMany`, and `updateByQuery` respectively 👍🏻  You should have a warning about that in your terminal too.

**Reply by gdpdils** [reply]
Hi and thank you. Can you point me out to the documentation for updateOne and updateByQuery? I cannot seem to find it where I'd expect it to be, here: https://docs.directus.io/reference/sdk/#update-multiple-items

Thanks.

**Reply by rijkvanzanten** [reply]
The API internal services for use in extension aren't documented yet, as we're still finalizing the exact final specs for them (as evident by this change from `update` to `updateOne` / `updateMany`. Your best bet is to take a peak through the types in the implementation 👍🏻 https://github.com/directus/directus/blob/main/api/src/services/items.ts#L49
